src/runtimes/node/zip_esbuild.js

```diff
@@ -22,6 +22,19 @@ const getAliases = ({ bundlePath, mainFile, sourcemapPath, srcDir }) => {
   return aliases
 }
 
+const getFunctionBasePath = ({ basePathFromConfig, mainFile, supportingSrcFiles }) => {
+  // If there is a base path defined in the config, we use that.
+  if (basePathFromConfig !== undefined) {
+    return basePathFromConfig
+  }
+
+  // If not, the base path is the common path prefix between all the supporting
+  // files and the main file.
+  const dirnames = supportingSrcFiles.map((filePath) => normalize(dirname(filePath)))
```

Comment:
```suggestion
  const dirnames = [...supportingSrcFiles, mainFile].map((filePath) => normalize(dirname(filePath)))
```

[sand] Since `mainFile` has uses the `dirname()` + `normalize()`